delivery_carrier_label_paazl/models/stock_picking.py

```diff
@@ -67,15 +67,26 @@ def _paazl_generate_labels(self):
         """
         self.ensure_one()
         api = self._paazl_api()
-        # we need to create labels in order to get a tracking number
-        status = api.service.generateLabels(**self._paazl_generate_labels_data())
+        is_return = self.location_id.usage == "customer"
+        return_rma = all([is_return and self.picking_type_code == "incoming"])
+        if return_rma:
+            payload = self._paazl_generate_labels_data()
+            # Removing since generatePdfReturnLabels call doesn't accept labelFormat
+            if "labelFormat" in payload:
+                payload.pop("labelFormat")
+            # Generating return labels
+            status = api.service.generatePdfReturnLabels(**payload)
+        else:
+            # we need to create labels in order to get a tracking number
+            status = api.service.generateLabels(**self._paazl_generate_labels_data())
         self._paazl_raise_error(status)
         # as we have the labels anyways, attach them
         account = self._get_carrier_account()
+        status_label = status.label if return_rma else status.labels
```

Comment:
Is this confirmed? It says in the PDF that I have that the generateReturnLabels request also returns `labels`
![image](https://user-images.githubusercontent.com/1033124/171858100-e6ceb41d-6597-458e-8997-82a966d67ffc.png)

Reply:
Yea I know, but reponse was geting .label for me after I was making calls to generatePdfReturnLabels.

Reply:
Alright, that settles it :grin: